### PR TITLE
fix: preserve CompactStorage benchmark evidence

### LINKED ISSUE #3894: M4 blocker: correct CompactStorage evidence timing and rewrite-plan attribution
Parent: #3732
Final evidence owner: #3737
Follow-up to merged PR #3886

## Blocker

Exact-head Codex review completed after #3886 merged and identified two unresolved evidence-integrity defects:

1. `leaf_pack.directory_sync_time_nanos` is populated from `DirectorySyncWaitTimeNanos`, although the name promises the full `DirectorySyncTimeNanos` operation duration.
2. Successful value-log rewrites overwrite `CompactStorageStats.ValueLogRewritePlan` with the final post-apply audit, so M0 artifacts can report zero selected plan bytes/segments while `SourceSegmentsRequested` proves that a rewrite ran.

These defects do not change production CompactStorage behavior, but they can misattribute phase cost and erase selected-work evidence. #3737 cannot certify the graph while the latest-head review findings remain unresolved.

## Scope

- Preserve the pre-apply `CompactStoragePlan` value-log rewrite plan when constructing each M0 measurement.
- Record full directory-sync operation time under `directory_sync_time_nanos`.
- Record non-overlapping directory-sync wait separately under an unambiguous additive field.
- Update deterministic schema/measurement tests and regenerate one representative rewrite artifact.
- Resolve both PR #3886 review threads by linking the follow-up PR.

## Test-first implementation

1. Add a failing unit test with distinct directory operation and wait durations; assert both survive under distinct JSON/struct fields and cannot be interchanged.
2. Add a failing unit test where the pre-apply plan selects one segment but final stats contain a zero post-apply plan; assert artifact plan fields retain the selected segment and byte totals while execution counters come from final stats.
3. Introduce the smallest measurement-constructor/API change that passes both tests.
4. Run focused tests, the M0 benchmark smoke fixture, race coverage for the touched package, and current required CI.

## Acceptance gates

- `directory_sync_time_nanos == sum(DirectorySyncTimeNanos)` for all pack runs.
- the separate wait field equals `sum(DirectorySyncWaitTimeNanos)`.
- a successful rewrite artifact reports pre-apply selected segments/total/live/stale bytes exactly and final source/copy/GC/debt counters exactly.
- deterministic JSON/schema tests pass; no production package/API behavior changes.
- focused `GOWORK=off go test ./TreeDB/db -run 'CompactStorageM0|CompactStorageMeasurement' -count=1` and race tests pass.
- one `full-high-debt` smoke artifact satisfies `plan_segments > 0`, `source_segments > 0`, and nonnegative timing/counter invariants.
- latest-head CI and mature AI review are green with zero unresolved threads before merge.

### PR DESCRIPTION
**Pull request by snissn**
## Summary

- record full directory-sync operation time under `directory_sync_time_nanos`
- add `directory_sync_wait_nanos` for the non-overlapping publication-path charge
- preserve the pre-apply value-log rewrite plan while retaining final execution/GC/debt counters

This resolves both exact-head Codex findings posted after #3886 merged.

## Test-first evidence

Red commit `7b4dcf857` added a regression that failed because the pre-apply-aware constructor did not exist. Implementation commit `d504156c6` makes it pass.

- `GOWORK=off go test ./TreeDB/db -run 'CompactStorageM0|CompactStorageMeasurement' -count=1`
- `GOWORK=off GOMAXPROCS=2 go test -race ./TreeDB/db -run 'CompactStorageM0|CompactStorageMeasurement' -count=1`
- `GOWORK=off GOMAXPROCS=2 GOMEMLIMIT=8GiB TREEDB_COMPACT_STORAGE_M0_SAMPLE=1 TREEDB_COMPACT_STORAGE_M0_ARTIFACT_DIR=/mnt/fast4tb/compact-storage-3894-smoke-d504156c6 taskset -c 0,1 go test ./TreeDB/db -run '^$' -bench '^BenchmarkCompactStorageM0/full-high-debt$' -benchtime=1x -count=1 -benchmem`

The smoke artifact reports one planned and requested source segment with `3,293,184` selected/requested bytes (`1,093,632` live, `2,199,552` stale); invariant checks passed.

Fixes #3894
Refs #3737
Refs #3732

### CODE REVIEW

**Comment by snissn** [comment]
@codex review

File: TreeDB/db/compact_storage_m0_measurement_test.go [hunk 1]
```diff
@@ -207,7 +221,8 @@ func newCompactStorageMeasurement(fixture compactStorageMeasurementFixture, arti
 		m.LeafPack.RelocationPlanTimeNanos += run.Pack.ApplyStages.RelocationTimeNanos
 		m.LeafPack.PageSyncTimeNanos += run.Pack.ApplyStages.PageSyncTimeNanos
 		m.LeafPack.CopiedLeafSyncTimeNanos += run.Pack.ApplyStages.LeafSyncTimeNanos
-		m.LeafPack.DirectorySyncTimeNanos += run.Pack.ApplyStages.DirectorySyncWaitTimeNanos
+		m.LeafPack.DirectorySyncTimeNanos += run.Pack.ApplyStages.DirectorySyncTimeNanos
```
[the remark this answers is not in the document]
Comment (snissn) [reply TreeDB/db/compact_storage_m0_measurement_test.go:226]: Fixed in f6fcd7e41. CompactStorage M0 artifacts are now schema v2, the capture environment records the expected version, reused/mixed run directories fail closed before aggregation, and docs define the operation-versus-wait timing semantics. Focused tests, race, shell syntax, and a real schema-v2 high-debt artifact pass.

**Comment by snissn** [comment]
@codex review

**Comment by snissn** [comment]
Exact-head re-review request for `3eb383bdb64e542aded59cc5244724bbc1a34169`, after updating onto current main `bea89acd43384b322afe59d3468288ee1a9f4779`. The five-file evidence-only diff is unchanged: schema v2 separates full directory-sync operation time from publication wait and preserves the pre-apply rewrite plan alongside final execution counters. Zero review threads are unresolved; fresh CI is running.\n\n@codex review
